### PR TITLE
[create-unimodule] Generalize android creation of android unimodule. …

### PR DESCRIPTION
**Pull request by mczernek**
…Do not rely on hardcoded path.

### CODE REVIEW

File: packages/expo-cli/src/commands/generate-module/configureModule.ts [hunk 1]
```diff
@@ -159,43 +159,37 @@ async function configureAndroid(
...
+    const destinationPath = sourcePath.replace(templatePackagePath, targetPackagePath);
+
+    await fse.mkdirp(destinationPath);
+    for (let file of sourceFilesPaths[sourcePath]) {
+      await fse.copy(
+        path.join(sourcePath, file),
+        path.join(destinationPath, file.replace('ModuleTemplate', moduleName))
+      );
+      await fse.remove(path.join(sourcePath, file));
+    }
```
Comment (bbarthec) [inline packages/expo-cli/src/commands/generate-module/configureModule.ts:187]:
```suggestion
  for (let [sourcePath, files] of Object.entries(sourceFilesPaths)) {
    const destinationPath = sourcePath.replace(templatePackagePath, targetPackagePath);

    await fse.mkdirp(destinationPath);
    for (let file of files) {
      await fse.copy(
        path.join(sourcePath, file),
        path.join(destinationPath, file.replace('ModuleTemplate', moduleName))
      );
      await fse.remove(path.join(sourcePath, file));
    }
```
That's my preference 💅 😉

File: packages/expo-cli/src/commands/generate-module/configureModule.ts [hunk 2]
```diff
@@ -299,6 +304,64 @@ async function configureNPM(
...
+ * @param result directories mapped to array containing its files
+ */
+export async function allFilesWithPathContaining(dirPath: string, containing: string) {
+  const all = await readAllFilesIn(dirPath);
+  const keys = Object.keys(all).filter(key => key.includes(containing));
+  const result: { [key: string]: string[] } = {};
+  for (let key of keys) {
+    result[key] = all[key];
+  }
+  return result;
```
Comment (bbarthec) [inline packages/expo-cli/src/commands/generate-module/configureModule.ts:320]:
```suggestion
  return Object.entries(await readAllFilesIn(dirPath))
  	.filter(([key]) => key.includes(containing))
  	.reduce<Record<string, string[]>>((acc, [key, value]) => {
  	  acc[key] = value;
  	  return acc;
  	}, {});
```
What would you say? 😄 💅

File: packages/expo-cli/src/commands/generate-module/configureModule.ts [hunk 3]
```diff
@@ -299,6 +304,64 @@ async function configureNPM(
...
+
+/**
+ * Returns a map of directiories mapped to its files, recursively from directory passed as @param dirPath
+ * @param dirPath Path to scan recursively
+ * @param result directories mapped to array containing its files
+ */
+async function readAllFilesIn(
+  dirPath: string,
+  result: { [key: string]: string[] } = {}
+): Promise<{ [key: string]: string[] }> {
```
Comment (bbarthec) [inline packages/expo-cli/src/commands/generate-module/configureModule.ts:331]:
```suggestion
async function readAllFilesIn(
  dirPath: string,
  result: Record<string, string[]> = {}
): Promise<Record<string, string[]>> {
```

**Comment by EvanBacon** [comment]
generate-module is deprecated

**Pull request closed without merging**